src/verify.integration.test.ts

```diff
@@ -2,324 +2,268 @@
  * @jest-environment node
  */
 
-import verify, { verifyWithIndividualChecks } from "./index";
-import { documentMainnetValid } from "../test/fixtures/documentMainnetValid";
-import { documentTampered } from "../test/fixtures/tampered-document";
-import { documentRopstenValid } from "../test/fixtures/documentRopstenValid";
-import { tokenRopstenValid } from "../test/fixtures/tokenRopstenValid";
-import { tokenRopstenInvalid } from "../test/fixtures/tokenRopstenInvalid";
+import { defaultVerificationManager } from "./index";
+import { documentMainnetValidWithCertificateStore } from "../test/fixtures/documentMainnetValidWithCertificateStore";
+import { documentTamperedWithCertificateStore } from "../test/fixtures/tampered-document";
+import { documentRopstenValidWithCertificateStore } from "../test/fixtures/documentRopstenValidWithCertificateStore";
+import { documentRopstenValidWithToken } from "../test/fixtures/documentRopstenValidWithToken";
+import { documentRopstenRevokedWithToken } from "../test/fixtures/documentRopstenRevokedWithToken";
 
 describe("verify(integration)", () => {
-  it("returns false if document's hash is invalid and was not issued", async () => {
-    const results = await verify(documentTampered, "ropsten");
+  it("should for OpenAttestationTamperCheck and OpenAttestationEthereumDocumentStoreIssued when document's hash is invalid and was not issued", async () => {
+    const results = await defaultVerificationManager(documentTamperedWithCertificateStore, {
+      network: "ropsten"
+    });
 
-    expect(results).toEqual(
-      expect.objectContaining({
-        hash: { checksumMatch: false },
-        issued: {
-          issuedOnAll: false,
+    expect(results).toStrictEqual([
+      {
+        data: {
+          checksumMatch: false
+        },
+        message: "Certificate has been tampered with",
+        status: "INVALID",
+        type: "OpenAttestationTamperCheck"
+      },
+      {
+        data: {
           details: [
             {
               address: "0x20bc9C354A18C8178A713B9BcCFFaC2152b53990",
-              error: expect.stringMatching("exception"),
+              error:
+                'call exception (address="0x20bc9C354A18C8178A713B9BcCFFaC2152b53990", method="isIssued(bytes32)", args=["0x85df2b4e905a82cf10c317df8f4b659b5cf38cc12bd5fbaffba5fc901ef0011b"], version=4.0.40)',
               issued: false
             }
-          ]
+          ],
+          issuedOnAll: false
         },
-        revoked: {
-          revokedOnAny: false,
+        message: "Certificate has not been issued",
+        status: "INVALID",
+        type: "OpenAttestationEthereumDocumentStoreIssued"
```

Comment:
Am thinking that the type should be succinct and general, serving more as a class of verification. We can put these as `name` of the verification method instead.

Valid class of checks: 
`INTEGRITY` class verification methods checks if the document has been tampered with.
`ISSUANCE_STATUES` class verification methods checks if the document has been issued and is in the right status (ie. not revoked)
`ISSUER_IDENTITY` class verification methods checks for the issuer's identity.

In this case the default methods will be:
OpenAttestationTamperCheck -> `INTEGRITY`
OpenAttestationEthereumDocumentStoreIssued => `ISSUANCE_STATUES`
OpenAttestationEthereumDocumentStoreRevoked => `ISSUANCE_STATUES`
OpenAttestationDnsTxtIdentity => `ISSUER_IDENTITY`

Possible extension will look like: 
OpenAttestationEthereumTokenRegistryIssued => `ISSUANCE_STATUES`
OpenCertsIdentityRegistry  => `ISSUER_IDENTITY`
SomeCountryIdentityRegistry  => `ISSUER_IDENTITY`
CustomExpiryCheck => `ISSUANCE_STATUES`

Using a known class, it can allow client application to check what type of checks has been performed it they like.

Reply:
done